internal/cache/option.go

```diff
@@ -36,52 +35,45 @@ var (
 )
 
 func WithExpiredHook(f func(context.Context, string)) Option {
-	return func(c *cache) error {
+	return func(c *cache) {
 		if f != nil {
 			c.expiredHook = f
 		}
-		return nil
 	}
 }
 
 func WithType(mo string) Option {
-	return func(c *cache) error {
+	return func(c *cache) {
 		if len(mo) == 0 {
```

Comment:
@hlts2 which one do you think is better?
```suggestion
		if mo != "" {
```

Reply:
I think `len(mo) == 0` is better.

Reply:
I'm not particular about it, so I think either one is fine.
but uber use `len() == 0` in to the `[style guide](https://github.com/uber-go/guide/blob/master/style.md#dont-panic)

Reply:
thank you.